modules/sagemaker/sagemaker-templates-service-catalog/templates/xgboost_abalone/product_stack.py

```diff
@@ -166,10 +174,16 @@ def __init__(
                         "sagemaker:UpdateModelPackage",
                         "sagemaker:CreateModel",
                     ],
-                    resources=[model_package_arn],
+                    resources=[
+                        (
+                            f"arn:{Aws.PARTITION}:sagemaker:{Aws.REGION}:{Aws.ACCOUNT_ID}:model-package/"
+                            f"{model_package_group_name}/*"
+                        )
+                    ],
                     principals=[
-                        iam.AccountPrincipal(pre_prod_account_id),
-                        iam.AccountPrincipal(prod_account_id),
+                        iam.ArnPrincipal(f"arn:aws:iam::{dev_account_id}:root"),
+                        iam.ArnPrincipal(f"arn:aws:iam::{pre_prod_account_id}:root"),
+                        iam.ArnPrincipal(f"arn:aws:iam::{prod_account_id}:root"),
```

Comment:
Minor nitpick but I wonder why you're not using [AccountRootPrincipal](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.AccountRootPrincipal.html) here. If there's a good reason why you need to use `ArnPrincipal`, please use `self.partition` instead of `"aws"`. I think AccountRootPrincipal does this automatically.

Same thing for `DescribeModelPackageGroup` as well.